plugins/aws/sts_provisioner.go

```diff
@@ -355,3 +360,30 @@ func getSTSClient(region string, credsProvider aws.CredentialsProvider) *sts.Cli
 	}
 	return sts.NewFromConfig(clientConfig)
 }
+
+func DetectSourceProfileLoop(profile string, config *confighelpers.ConfigFile) error {
+	visited := make(map[string]bool)
+	sourceProfile := profile
+
+	for sourceProfile != "" {
+		if visited[sourceProfile] {
+			return fmt.Errorf("infinite loop in credential configuration detected. Attempting to load from profile %q which has already been visited", sourceProfile)
+		} else {
+			visited[sourceProfile] = true
+		}
+
+		profileSection, ok := config.ProfileSection(sourceProfile)
+		if !ok {
+			return fmt.Errorf("source profile %s does not exist in your AWS config file", sourceProfile)
```

Comment:
The profile is user input, so let's use `%q` here.
```suggestion
			return fmt.Errorf("source profile %q does not exist in your AWS config file", sourceProfile)
```

Reply:
Done, thanks!